packages/vuetify/src/components/VCalendar/mixins/calendar-with-events.ts

```diff
@@ -0,0 +1,475 @@
+
+type VNameFunction = (event: VEventParsed, timedEvent: boolean) => string
+
+type VTimeToY = (time: VTimestamp | number | string) => number
+
+type VEventResetCheck = (date: VTimestamp) => void
+
+type VEventVisualGetOffset = (visual: VEventVisual, visuals: VEventVisual[]) => number
+
+type VEventGetter = (day: VTimestamp) => VEventParsed[]
+
+type VEventVisualGetter = (events: VEventParsed[], timed: boolean) => VEventVisual[]
+
+type VEventVisualToNode<D> = (visual: VEventVisual, index: number, day: D) => VNode
+
+type VDailyEventsMap = {
+  [date: string]: {
+    parent: HTMLElement
+    more: HTMLElement | null
+    events: HTMLElement[]
+  }
+}
+
+interface VEventVisual {
+  offset: number
+  event: VEventParsed
+  columnCount: number
+  column: number
+}
+
+interface VDaySlotScope extends VTimestamp {
+  outside: boolean
+  index: number
+}
+
+interface VDayBodySlotScope extends VDaySlotScope {
+  timeToY: VTimeToY
+}
+
+/* @vue/component */
+export default CalendarBase.extend({
+  name: 'calendar-with-events',
+
+  props: props.events,
+
+  computed: {
+    noEvents (): boolean {
+      return this.events.length === 0
+    },
+    parsedEvents (): VEventParsed[] {
+      return this.events.map((input, index) => parseEvent(input, index, this.eventStart, this.eventEnd))
+    },
+    eventColorFunction (): VColorFunction {
+      return typeof this.eventColor === 'function'
+        ? this.eventColor as VColorFunction
+        : () => (this.eventColor as string)
+    },
+    eventTextColorFunction (): VColorFunction {
+      return typeof this.eventTextColor === 'function'
+        ? this.eventTextColor as VColorFunction
+        : () => (this.eventTextColor as string)
+    },
+    eventNameFunction (): VNameFunction {
+      return typeof this.eventName === 'function'
+        ? this.eventName as VNameFunction
+        : (event, timedEvent) => {
+          const name = event.input[this.eventName as string] as string
+          if (event.start.hasTime) {
+            if (timedEvent) {
+              const showStart = event.start.hour < 12 && event.end.hour >= 12
+              const start = this.formatTime(event.start, showStart)
+              const end = this.formatTime(event.end, true)
+              return `<strong>${name}</strong><br>${start} - ${end}`
+            } else {
+              const time = this.formatTime(event.start, true)
+              return `<strong>${time}</strong> ${name}`
+            }
+          }
+          return name
+        }
+    },
+  },
+
+  methods: {
+    formatTime (withTime: VTimestamp, ampm: boolean): string {
+      const suffix = ampm ? (withTime.hour < 12 ? 'a' : 'p') : ''
+      const hour = withTime.hour % 12 || 12
+      const minute = withTime.minute
+
+      return minute > 0
+        ? (minute < 10
+          ? `${hour}:0${minute}${suffix}`
+          : `${hour}:${minute}${suffix}`)
+        : `${hour}${suffix}`
+    },
+    updateEventVisibility (): void {
```

Comment:
IMHO no need for void types
```suggestion
    updateEventVisibility () {
```